astropy/cosmology/funcs.py

```diff
@@ -34,20 +35,49 @@ def z_at_value(func, fval, zmin=1e-8, zmax=1000, ztol=1e-8, maxfun=500):
     ----------
     func : function or method
        A function that takes a redshift as input.
-    fval : astropy.Quantity instance
-       The value of ``func(z)``.
+
+    fval : `~astropy.units.Quantity` instance
+       The (scalar) value of ``func(z)`` to recover.
+
     zmin : float, optional
        The lower search limit for ``z``.  Beware of divergences
        in some cosmological functions, such as distance moduli,
        at z=0 (default 1e-8).
+
     zmax : float, optional
        The upper search limit for ``z`` (default 1000).
+
     ztol : float, optional
        The relative error in ``z`` acceptable for convergence.
+
     maxfun : int, optional
        The maximum number of function evaluations allowed in the
        optimization routine (default 500).
 
+    method : str or callable, optional
+       Type of solver to pass to ``~scipy.optimize.minimize_scalar`` -
+       should be one of 'Brent' (default), 'Golden' or 'Bounded'.
+       Can in theory also be a callable object as custom solver,
+       but this is untested.
```

Comment:
Maybe a little more verbose, e.g. about case insensitivity and how to implement custom solvers.

```suggestion
       Type of solver to pass to ``~scipy.optimize.minimize_scalar``.
       The built-in options are 'Brent' (default), 'Golden', or 'Bounded',
       and are case insensitive. In principle, custom solvers can also be used
       by passing any callable object that follows the requirements explained
       in section "Custom minimizers" of :fun:`scipy.optimize.minimize_scalar`.
       The use here of custom solvers is currently untested.
```

Reply:
Yea, I just took that from the scipy docstring, which is just as sloppy about capitalisation.
I've been rather ambivalent about advertising the custom solvers as I did not really find good usage examples, and I guess it's rather unlikely that cosmology users want to bring along their own (at least for now).
But since it is an option in `minimize_scalar` one should probably not hide it. Definitely better with that bit of extra explanation, thanks!
Actually the best documentation I found is probably the tutorial entry https://docs.scipy.org/doc/scipy/reference/tutorial/optimize.html#custom-minimizers , but is this something that could be linked with Xref?

Reply:
That link would be good to include directly. It's funny that the least important feature requires the most explanation.
For something using intersphinx docs reference (https://docs.readthedocs.io/en/stable/guides/intersphinx.html#using-intersphinx ), maybe something along the lines of.

```In principle, custom solvers can also be used, though their use here is currently untested. See section "Custom minimizers" of :fun:`scipy.optimize.minimize_scalar` or of :doc:`scipy:reference/tutorial/optimize` for details.```

Reply:
This appears resolved. 👍